src/Microsoft.VisualStudio.Composition/RuntimeComposition.cs

```diff
@@ -21,12 +21,22 @@ public class RuntimeComposition : IEquatable<RuntimeComposition>
         private readonly IReadOnlyDictionary<string, IReadOnlyCollection<RuntimeExport>> exportsByContractName;
         private readonly IReadOnlyDictionary<TypeRef, RuntimeExport> metadataViewsAndProviders;
 
+        public const string InvalidCompositionException = nameof(InvalidCompositionException);
+
         private RuntimeComposition(IEnumerable<RuntimePart> parts, IReadOnlyDictionary<TypeRef, RuntimeExport> metadataViewsAndProviders, Resolver resolver)
         {
             Requires.NotNull(parts, nameof(parts));
             Requires.NotNull(metadataViewsAndProviders, nameof(metadataViewsAndProviders));
             Requires.NotNull(resolver, nameof(resolver));
 
+            if (!parts.Any() || !metadataViewsAndProviders.Any())
+            {
+                var exception = new ArgumentException("Invalid arguments passed to generate runtime composition.");
```

Comment:
Can we separate the `parts.Any()` and `metadataViewAndProviders.Any()` checks and then throw an exception that actually calls out which argument is invalid? When users hit this (and possibly when _we_ hit this) it helps diagnosis when we know what exactly is wrong.